### PR TITLE
Support defining column names in CTAS

### PR DESCRIPTION
**Pull request by douenergy**
Implementing support for explicit column name definition in `CREATE TABLE AS` (CTAS) statements 

```sql 
CREATE TABLE tbl(col1, col2) AS 
SELECT 1, 'hello';
```
| col1 | col2  |
|-----:|-------|
| 1    | hello |

Additionally, this implementation adheres to PostgreSQL behavior when column name definitions and query result counts do not match.
```sql
CREATE TABLE tbl(col1) AS 
SELECT 4 ,'mismatch';
```
| col1 | 'mismatch' |
|-----:|------------|
| 4    | mismatch   |

When the number of defined column names is less than the query result columns, we first use the specified column names, then adopt the query result column names for the remaining columns, starting from where the defined names left off.

```sql
CREATE TABLE tbl(col1, col2) 
AS SELECT 5;
```
**Binder Error: Target table has more colum names than query result.**

### CODE REVIEW

File: src/catalog/catalog_entry/table_catalog_entry.cpp [hunk 1]
```diff
@@ -176,6 +176,24 @@ string TableCatalogEntry::ColumnsToSQL(const ColumnList &columns, const vector<u
...
+	}
+
+	std::stringstream ss;
+	ss << "(";
+
+	for (auto &column : columns.Logical()) {
+		if (column.Oid() > 0) {
+			ss << ", ";
+		}
+		ss << column.Name();
```
Comment (Mytherin) [inline src/catalog/catalog_entry/table_catalog_entry.cpp:191]: This is likely missing `KeywordHelper::WriteOptionallyQuoted` - can we add a test where we use column names that require quoting (e.g. `"Column Name"`)?

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! LGTM - one minor comment:

**Comment by doupache** [comment]
@Mytherin Thanks for reviewing. I added KeywordHelper::WriteOptionallyQuoted to write column names, and a corresponding test.

**Comment by doupache** [comment]
I would also like to work on  a CTAS extension as follows. 

```sql
CREATE TABLE tbl(col) AS VALUES
(1),
(2),
(3)
```

This following SQL is valid in both `DuckDB` and `Postgres`. 
```sql 
VALUES (1), (2),(3);
```
The VALUES clause serves as a legitimate simple SELECT statement. We can execute this query in both DuckDB and PostgreSQL without issues.


### grammar 
I propose adding the following rule to the create_as.y file
```
CREATE_P ... TABLE create_as_target AS values_clause_opt_comma opt_with_data
```


I've found a useful case for defining small tables. It's more intuitive and doesn't require jumping around in a SELECT query like normal CTAS does. For example:(compare to normal CTAS) e.g. 

**CTAS**
```sql 
CREATE TABLE tbl AS 
SELECT * FROM (
 VALUES (1),
(2),
(3)
) t(col)
```

**CTA VALUES**
```sql
CREATE TABLE tbl(col) AS VALUES
(1),
(2),
(3)
```


What do you think? If you agree this extension is a good idea, maybe we could continue this discussion in a new issue I'd like to open.

**Comment by douenergy** [comment]
I would also like to work on  a CTAS extension as follows. 

```sql
CREATE TABLE tbl(col) AS VALUES
(1),
(2),
(3)
```

This following SQL is valid in both `DuckDB` and `Postgres`. 
```sql 
VALUES (1), (2),(3);
```
The VALUES clause serves as a legitimate simple SELECT statement. We can execute this query in both DuckDB and PostgreSQL without issues.


### grammar 
I propose adding the following rule to the create_as.y file
```
CREATE_P ... TABLE create_as_target AS values_clause_opt_comma opt_with_data
```


I've found a useful case for defining small tables. It's more intuitive and doesn't require jumping around in a SELECT query like normal CTAS does. For example:(compare to normal CTAS) e.g. 

**CTAS**
```sql 
CREATE TABLE tbl AS 
SELECT * FROM (
 VALUES (1),
(2),
(3)
) t(col)
```

**CTA VALUES**
```sql
CREATE TABLE tbl(col) AS VALUES
(1),
(2),
(3)
```


What do you think? If you agree this extension is a good idea, maybe we could continue this discussion in a new issue I'd like to open.

**Comment by Mytherin** [comment]
> ```sql
> CREATE TABLE tbl(col) AS VALUES
> (1),
> (2),
> (3)
> ```

I don't quite follow, isn't this already supported? `VALUES` can be used whereever a normal `SELECT` can be used. This works for me in all systems I've tested (DuckDB/Postgres/SQLite):

```sql
CREATE TABLE tbl AS VALUES (1), (2), (3);

```

**Comment by douenergy** [comment]
@Mytherin Sorry for the confusion. I overlooked the part of the binder codebase. After this PR gets merged, I'll have everything I need. 😅

My original purpose is to define the column name tbl(col) and VALUES in one go.

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**